parsl/tests/test_htex/test_command_client_timeout.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+import threading
+import time
+import zmq
+from parsl import curvezmq
+from parsl.executors.high_throughput.zmq_pipes import CommandClient
+from parsl.executors.high_throughput.errors import CommandClientTimeoutError, CommandClientBadError
+
+
+# Time constant used for timeout tests: various delays and
+# timeouts will be appropriate multiples of this, but the
+# value of T itself should not matter too much as long as
+# it is big enough for zmq connections to happen successfully.
+T = 0.25
+
+
+@pytest.mark.local
+def test_command_not_sent() -> None:
+    """Tests timeout on command send.
+    """
+    ctx = curvezmq.ClientContext(None)
+
+    # RFC6335 ephemeral port range
+    cc = CommandClient(ctx, "127.0.0.1", (49152, 65535))
+
+    # cc will now wait for a connection, but we won't do anything to make the
+    # other side of the connection exist, so any command given to cc should
+    # timeout.
+
+    with pytest.raises(CommandClientTimeoutError):
```

Comment:
We may want to include a test for the `InternalConsistencyError` path as well.

Reply:
It's meant to be "impossible" in the sense of an assertion failure

Reply:
Makes sense; your call.